app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,61 @@
+class VendorsController < ApplicationController
+
+  def index
+    if params[:market_id] == nil
+      @vendors = Vendor.all
+    else
+      @vendors = Vendor.where(market_id: params[:market_id])
+      @market_edit = true
+    end
+  end
```

Comment:
Interesting way to deal with this problem! @market_edit is a very clear variable.